tests/posix/common/src/clock.c

```diff
@@ -72,8 +72,13 @@ void test_posix_realtime(void)
 	zassert_equal(ret, 0, "Fail to get realtime clock");
 
 	zassert_equal(rts.tv_sec, mts.tv_sec, "Seconds not equal");
-	zassert_equal(rts.tv_nsec, mts.tv_nsec, "Nanoseconds not equal");
 
+#ifdef CONFIG_BOARD_EHL_CRB
```

Comment:
Since you say the issue is with HPET, would it make more sense to test for `CONFIG_HPET_TIMER` instead of a specific board?

Reply:
HPET implementation can run at high frequency. IIRC, most PC HPET timer are running at 19.2MHz or 25MHz.

Reply:
please just exclude it in the testcase.yaml

Reply:
Thanks for asking. It seemed more an issue with EHL but also HPET timer so wasn't quite sure. Great to have feedback. 

But turns out this is not a favored way to approach this issue anyway. fwiw I was trying to preserve the rest of the test_posix_realtime() for EHL as well since we only had issue at a particular granularity and could still exercise most of the test otherwise. Alas, this is not correct so I've adapted to the feedback (moved to exclude the platform in testcase.yaml). Thanks all!